### ISSUE TITLE
pip install weather2 fails on Windows 11 with python 3.12

### ISSUE THREAD

**Issue by WiebeBaron** [jenca-adam/weather#6]
Get the following error:

```
Getting requirements to build wheel ... error
  error: subprocess-exited-with-error

  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
╰─> [24 lines of output]
      Traceback (most recent call last):
        File "...\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
          main()
        File "...\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "...\Python\Python312\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "...\Temp\pip-build-env-dcfzbjz7\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "...\Temp\pip-build-env-dcfzbjz7\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "...\Temp\pip-build-env-dcfzbjz7\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "...l\Temp\pip-build-env-dcfzbjz7\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 37, in <module>
        File "<string>", line 11, in read_description
        File "...\Python\Python312\Lib\encodings\cp1252.py", line 23, in decode
          return codecs.charmap_decode(input,self.errors,decoding_table)[0]
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 876: character maps to <undefined>
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Comment by jenca-adam** [comment]
Related: kcsaff/getkey#2

Apparently the `getkey` module `weather` uses is no longer maintained. However, there is a fork (https://github.com/TheIceBergBot/getkey) that has the PyPI package `get-key`. I will set it as a requirement in `setup.py`.

Let me know if this fixes your problem, as I don't have access to Windows

**Comment by WiebeBaron** [comment]
Thanks for getting back so quickly, that has indeed resolved the install problem.

It looks like there may still be a windows related issue, I get the following error (also using CLI):

```
import weather
forecast = weather.forecast(debug=True)
```

---
INFO:Debugger has started
INFO:running service "yrno"
INFO:requested url

Exception has occurred: FileNotFoundError
[WinError 3] The system cannot find the path specified: '.cache/weather/fc/xml/New Zealand / Aotearoa'
  File "C:\dev\python\MyTest.py", line 6, in <module>
    forecast = weather.forecast(debug=True)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [WinError 3] The system cannot find the path specified: '.cache/weather/fc/xml/New Zealand / Aotearoa'

---

One more small thing is that 'source' from the readme is now 'service' in the weather.forecast definition.

Thanks!

**Comment by jenca-adam** [comment]
Can you provide the full code so that i can reproduce your issue?

**Comment by WiebeBaron** [comment]
That is the full source, just the import and the forecast.

Below is the output if calling weather from the command line:

weather --debug
←[36m\\Python\Python312\Lib\site-packages\weather\__init__.py:1726:forecast ←[39m-> ←[94mINFO←[0m:←[94mDebugger has started←[0m
←[36m\\Python\Python312\Lib\site-packages\weather\__init__.py:1634:forecast ←[39m-> ←[94mINFO←[0m:←[94mrunning service "yrno"←[0m
←[36m\\Python\Python312\Lib\site-packages\weather\__init__.py:134:request ←[39m-> ←[94mINFO←[0m:←[94mrequested url←[0m
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "\\Python\Python312\Scripts\weather.exe\__main__.py", line 7, in <module>
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 1801, in main
    foc=forecast(args.city[0],args.country[0],service=args.service,debug=args.debug)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 1728, in forecast
    return service.forecast(cityname,countryname,unit)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 1636, in forecast
    ress.append( SERVICES[service].forecast(*a,**k))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 1099, in forecast
    return self.ForecastParser.parse(BetaChrome().request(apiurl)[1],unit,cityname,countryname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 1030, in parse
    cacher.cache(foc)
  File "\\Python\Python312\Lib\site-packages\weather\__init__.py", line 474, in cache
    os.makedirs(pt,exist_ok=True)
  File "<frozen os>", line 215, in makedirs
  File "<frozen os>", line 225, in makedirs
FileNotFoundError: [WinError 3] The system cannot find the path specified: '.cache/weather/fc/xml/New Zealand / Aotearoa'

**Comment by jenca-adam** [comment]
This seems to be an issue with Nominatim returning the country name for New Zealand in multiple languages. I tried to fix this and it seems to work, but if you had any further problems, feel free to create a new issue.

**Comment by WiebeBaron** [comment]
I tried changing the location by using

`forecast=weather.forecast('New York')`

and still got this error:

`FileNotFoundError: [WinError 3] The system cannot find the path specified: '.cache/weather/fc/xml/New Zealand / Aotearoa' `

So I tried just creating that directory (Aotearoa was missing) and it now works!  NZ is often called (New Zealand / Aotearoa), that might be causing directory issues?

Thanks for looking at this!
